lib/maths/CLogNormalMeanPrecConjugate.cc

```diff
@@ -524,9 +523,15 @@ class CLogMarginalLikelihood : core::CNonCopyable {
             double impliedPrecision = m_Precision + weightedNumberSamples;
 
             m_Constant = 0.5 * (std::log(m_Precision) - std::log(impliedPrecision)) -
-                         0.5 * m_NumberSamples * LOG_2_PI - 0.5 * logVarianceScaleSum +
-                         boost::math::lgamma(impliedShape) -
-                         boost::math::lgamma(m_Shape) + m_Shape * std::log(m_Rate);
+                         0.5 * m_NumberSamples * LOG_2_PI -
+                         0.5 * logVarianceScaleSum + std::lgamma(impliedShape) -
+                         std::lgamma(m_Shape) + m_Shape * std::log(m_Rate);
+
+            if (std::isfinite(m_Constant) == false) {
```

Comment:
As above: check for NaN and no need for try catch.

Reply:
as above, I use `isfinite()` because it combines the check for infinite and NaN